app/controllers/subgroups_controller.rb

```diff
@@ -3,25 +3,54 @@ class SubgroupsController < ApplicationController
   before_action :set_group
   before_action :set_signup_mode
   before_action :set_target_action
+  before_action :set_oauth, only: %i[oauth_signup create]
 
   # GET groups/:group_id/subgroups/new
   def new
     @subgroup = Group.build_group_with_organizer
   end
 
-  # GET groups/:group_id/subgroups/signup
+  # POST groups/:group_id/subgroups/signup
   def signup
     @person = Person.build_for_signup
     @subgroup = JSON.generate(subgroup_params.to_h)
-    render "signup_form_#{@signup_mode}", layout: 'signup'
+    if @signup_mode == 'email'
+      render "signup_form_#{@signup_mode}", layout: 'signup'
+    else
+      redirect_to omniauth_authorize_path(
+        :person,
+        omniauth_value,
+        subgroup_json: JSON.generate(subgroup_params.to_h),
+        auth_action:   'signup',
+        signup_reason: 'create_group',
+        group_id: @group.id
+      )
+    end
+  end
+
+  # GET groups/:group_id/subgroups/oauth_signup
```

Comment:
maybe one more comment to make clear why we do this? (so that we don't GET and POST to same endpoint?)